File: docs/app/Button.js
import { AbstractDrawable } from "./AbstractDrawable.js";
import { COLOR } from "./Constants.js";
import { Rect } from "./Rectangle.js";
import { Text } from "./Text.js";

export class Button extends AbstractDrawable {
    constructor(x = null, y = null, w = null, h = null, text="", padding = 0) {
        super(x, y + 3, w, h);

        this.text = text;
        
        this.textIconColor = COLOR.BLACK;

        this.buttonRect = new Rect(x, y, w, h, undefined, 5, false);
        this.textIcon = new Text(x + padding,y, w, h, text, this.textIconColor);
    }

    DrawObject(screen) {
        this.buttonRect.DrawObject(screen);
        this.textIcon.DrawObject(screen);
    }

    IsInside(position){
        return this.x <= position[0] && this.x + this.w >= position[0] && this.y <= position[1] && this.y + this.h >= position[1];
    }

    OnClick(eventType, context) {
        console.log("Unbound button");
    }

    SetAccentColor(accent) {
        this.buttonRect.SetAccentColor(accent);
    }

    SetTextIconColor(color){
        this.textIconColor = color;
    }

    SelectThis() {
        this.buttonRect.SetColor(COLOR.GREEN);
    }

    DeselectThis() {
        this.buttonRect.SetColor(COLOR.BLACK);
    }
}

File: mis-python/App/Button.py
# -*- coding: utf-8 -*-

from App.AbstractDrawable import AbstractDrawable
from App.Rectangle import Rect
from App.Constants import COLOR as Color
from App.Text import Text


class Button(AbstractDrawable):

    def __init__(self, x=None, y=None, w=None, h=None, text="", padding=0):
        super(Button, self).__init__(x=x, y=y, w=w, h=h)

        self.text = text

        self.text_icon_color = Color.BLACK

        self.button = Rect(x , y , w , h, width=5, transparent=False)
        self.texticon = Text(x + padding, y, w, h, text, self.text_icon_color)

    def DrawObject(self, screen):
        self.button.DrawObject(screen)
        self.texticon.DrawObject(screen)

    def IsInside(self, position):
        return self.x <= position[0] <= self.x + self.w and self.y <= position[1] <= self.y + self.h

    def bind(self, funcpointer):
        self.OnClick = funcpointer

    def OnClick(self, event_type, context):
        raise RuntimeError("Undobund button")

    def SetAccentColor(self, accent):
        self.button.SetAccentColor(accent)

    def SelectThis(self):
        self.button.SetColor(Color.GREEN)

    def DeselectThis(self):
        self.button.SetColor(Color.BLACK)